ext/zend_test/test.c

```diff
@@ -243,12 +243,16 @@ static ZEND_FUNCTION(zend_test_compile_string)
 
 	ZEND_PARSE_PARAMETERS_START(3, 3)
 		Z_PARAM_STR(source_string)
-		Z_PARAM_PATH_STR_EX(filename, 1, 0)
+		Z_PARAM_PATH_STR(filename)
 		Z_PARAM_LONG(position)
 	ZEND_PARSE_PARAMETERS_END();
 
 	zend_op_array *op_array = NULL;
 
+	if (zend_str_has_nul_byte(filename)) {
```

Comment:
Using PATH_STR makes this condition not necessary.

Reply:
but it seems PATH_STR does not do the check_null part. no ?

Reply:
check_null means checking for a NULL argument instead of a string, not for NUL bytes.
The bug here was that you passed 1 to check_null instead of 0. And passing 0 is equivalent to PATH_STR without the EX

Reply:
The whole reason we have PATH_STR is to check for NUL bytes in the string, that's the only difference with normal STR fast-ZPP